knotx-server/src/main/java/io/knotx/server/KnotxEngineHandler.java

```diff
@@ -93,4 +97,10 @@ private void handleRoute(final RoutingContext context, final String address,
         );
   }
 
+  private KnotProxy getProxyWithOptions(String address, DeliveryOptions deliveryOptions) {
```

Comment:
You may replace this method with:

`
proxies.computeIfAbsent(address, adr -> KnotProxy.createProxyWithOptions(vertx, adr, deliveryOptions));
`